src/main/kotlin/com/nikitanazarov/coroutinestacks/ui/CoroutineFramesList.kt

```diff
@@ -0,0 +1,132 @@
+        val lastStackFrame = data.getOrNull(1)
+
+        border = if (lastRunningStackFrame == lastStackFrame) {
+            createRoundedBorder(JBColor.BLUE)
+        } else {
+            createRoundedBorder(JBColor.BLACK)
+        }
+
+        cellRenderer = object : DefaultListCellRenderer() {
+            override fun getListCellRendererComponent(
+                list: JList<*>,
+                value: Any,
+                index: Int,
+                isSelected: Boolean,
+                cellHasFocus: Boolean
+            ): Component {
+                val renderer = super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus)
+                if (renderer !is JComponent) {
+                    return renderer
+                }
+
+                with(renderer) {
+                    val listSize = list.model.size
+                    if (index == 0) {
+                        toolTipText = trace.coroutinesActiveLabel
+                        font = font.deriveFont(Font.BOLD)
+                    } else if (index < listSize) {
+                        toolTipText = value.toString()
+                    }
+                    border = when {
+                        index < listSize - 1 -> compoundBorder
+                        index == listSize - 1 -> leftPaddingBorder
+                        else -> null
+                    }
+                }
+
+                return renderer
+            }
+        }
+
+        addMouseListener(object : MouseAdapter() {
+            override fun mouseClicked(e: MouseEvent?) {
+                val list = e?.source as? JBList<*> ?: return
+                val index = list.locationToIndex(e.point).takeIf { it > 0 } ?: return
+                val stackFrameItem = trace.stackFrameItems[index - 1]
+
+                stackFrameItem?.let { frameItem ->
+                    val frame = frameItem.createFrame(debugProcess)
+
+                    if (suspendContext.activeExecutionStack != null && frame != null) {
+                        suspendContext.setCurrentStackFrame(suspendContext.activeExecutionStack, frame)
+                    }
+                }
+            }
+        })
+    }
+
+    // Copied from org.jetbrains.kotlin.idea.debugger.coroutine.view.CoroutineSelectedNodeListener#setCurrentStackFrame
+    private fun SuspendContextImpl.setCurrentStackFrame(executionStack: XExecutionStack?, stackFrame: XStackFrame) {
+        val fileToNavigate = stackFrame.sourcePosition?.file ?: return
+        val session = debugProcess.session.xDebugSession ?: return
+        if (!ClsClassFinder.isKotlinInternalCompiledFile(fileToNavigate)) {
+            ApplicationManager.getApplication().invokeLater {
+                if (executionStack != null) {
+                    session.setCurrentStackFrame(executionStack, stackFrame, false)
+                }
+            }
+        }
+    }
+
+    private fun createRoundedBorder(color: JBColor): Border {
```

Comment:
Actually we don't need this function, we can replace it with plain anonymous object like this:
```
val borderColor = if (...) BLUE else GRAY
border = object : LineBorder(...) { ... } // Use borderColor here
```